knowledge_repo/app/models.py

```diff
@@ -234,26 +235,59 @@ class Vote(db.Model):
 
 
 @unique_constructor(
-    lambda username: username,
-    lambda query, username: query.filter(User.username == username)
+    lambda identifier: identifier,
+    lambda query, identifier: query.filter(User.identifier == identifier)
 )
-class User(db.Model):
+class User(db.Model, UserMixin):
     __tablename__ = 'users'
 
     id = db.Column(db.Integer, primary_key=True)
-    username = db.Column(db.String(500))
     created_at = db.Column(db.DateTime, default=func.now())
 
+    identifier = db.Column(db.String(500))  # Unique identifier across all login methods
+
+    username = db.Column(db.String(500))  # Username used to log in (may differ from identifier)
+    password = db.Column(db.String(500))  # Password for local logins
```

Comment:
I was thinking we would hash them using sha256 or something. I suppose we *could* encrypt them, now that we have secret key; if we want to be able to recover the underlying password in the future... but I think that hashing them is more standard practice. If we plan to encrypt, then it may be necessary to increase the size of the password field column; whereas with hashing, 500 is more than generous :).